packages/fmg-turbo-adjudicator/contracts/TurboAdjudicator.sol

```diff
@@ -7,26 +7,16 @@ contract TurboAdjudicator {
     using State for State.StateStruct;
 
     struct Authorization {
-        // *********************************************************
-        // WARNING
-        // -------
-        // This authorization does not prevent collisions.
-        // It only prevents griefing: if the participant signs
-        // this authorization, we assume they wish to withdraw
-        // the amount to the destination. Therefore, any front-
-        // runner who submits the transaction in lieu of the
-        // participant is simply saving them some gas.
-        // (If the amount were not part of the authorization,
-        // then a malicious actor could submit a withdrawal
-        // for a tiny amount, forcing alice to submit a new
-        // signature with an increased nonce. If they have to
-        // submit the amount that the participant authorized,
-        // then the participant can be satisfied with the front-
-        // runner's transaction, which causes the desired transfer.)
-        // *********************************************************
+        // Prevents front-running:
```

Comment:
I think this would probably be better described as "prevents replay attacks". Front-running is when someone sees your transaction and puts a different one in first (e.g. for a trade on an exchange). I don't think that's possible here.

Maybe it should say something like: "by including the address of the sender in the data to sign, we prevent a different party from replaying the transaction in the future".

Reply:
I guess it might be worth explicitly pointing out that the source is the address used in the channel, which could be different from the sender.

Reply:
Good points